pkg/steps/run_test.go

```diff
@@ -40,135 +44,282 @@ func (*fakeStep) Objects() []ctrlruntimeclient.Object { return nil }
 
 func (f *fakeStep) Provides() api.ParameterMap { return nil }
 
-func TestRunNormalCase(t *testing.T) {
-	root := &fakeStep{
-		name:      "root",
-		shouldRun: true,
-		requires:  []api.StepLink{api.ExternalImageLink(api.ImageStreamTagReference{Namespace: "ns", Name: "base", Tag: "latest"})},
-		creates:   []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceRoot)},
-	}
-	other := &fakeStep{
-		name:      "other",
-		shouldRun: true,
-		requires:  []api.StepLink{api.ExternalImageLink(api.ImageStreamTagReference{Namespace: "ns", Name: "base", Tag: "other"})},
-		creates:   []api.StepLink{api.InternalImageLink("other")},
-	}
-	src := &fakeStep{
-		name:      "src",
-		shouldRun: true,
-		requires:  []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceRoot)},
-		creates:   []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceSource)},
-	}
-	bin := &fakeStep{
-		name:      "bin",
-		shouldRun: true,
-		requires:  []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceSource)},
-		creates:   []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceBinaries)},
-	}
-	testBin := &fakeStep{
-		name:      "testBin",
-		shouldRun: true,
-		requires:  []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceSource)},
-		creates:   []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceTestBinaries)},
-	}
-	rpm := &fakeStep{
-		name:      "rpm",
-		shouldRun: true,
-		requires:  []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceBinaries)},
-		creates:   []api.StepLink{api.InternalImageLink(api.PipelineImageStreamTagReferenceRPMs)},
-	}
-	unrelated := &fakeStep{
-		name:      "unrelated",
-		shouldRun: true,
-		requires:  []api.StepLink{api.InternalImageLink("other"), api.InternalImageLink(api.PipelineImageStreamTagReferenceRPMs)},
-		creates:   []api.StepLink{api.InternalImageLink("unrelated")},
-	}
-	final := &fakeStep{
-		name:      "final",
-		shouldRun: true,
-		requires:  []api.StepLink{api.InternalImageLink("unrelated")},
-		creates:   []api.StepLink{api.InternalImageLink("final")},
+func TestStepsRun(t *testing.T) {
+	testCases := []struct {
+		id          string
```

Comment:
the fact that the name attribute of your testcases is called `id` and not `name` is btw a good way to universally identify testcases originally written by you :)

Reply:
That's my signature :P